src/composable/orderTypes/Twap.ts

```diff
@@ -350,11 +350,37 @@ export class Twap extends ConditionalOrder<TwapData, TwapStruct> {
     const { numberOfParts } = this.data
     const startTimestamp = await this.startTimestamp(params)
 
+    if (blockTimestamp < startTimestamp) {
+      return {
+        result: PollResultCode.UNEXPECTED_ERROR,
+        reason: `TWAP part hash't started. First TWAP part start at ${startTimestamp} (${formatEpoch(startTimestamp)})`,
+        error: undefined,
+      }
+    }
+    const expireTime = numberOfParts.mul(timeBetweenParts).add(startTimestamp).toNumber()
+    if (blockTimestamp >= expireTime) {
+      return {
+        result: PollResultCode.UNEXPECTED_ERROR,
+        reason: `TWAP is expired. Expired at ${expireTime} (${formatEpoch(expireTime)})`,
+        error: undefined,
+      }
+    }
+
     // Get current part number
-    const currentPartNumber = Math.floor(blockTimestamp - startTimestamp / timeBetweenParts)
+    const currentPartNumber = Math.floor((blockTimestamp - startTimestamp) / timeBetweenParts)
```

Comment:
Disregard my comments in the PR #168  about this math, I see it's fixed here :+1: :guitar: